frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer.tsx

```diff
@@ -192,9 +212,14 @@ function MainNavbarContainer({
     const nonPersonalOrArchivedCollections = collections.filter(
       nonPersonalOrArchivedCollection,
     );
+    const nonAppsCollections = nonPersonalOrArchivedCollections.filter(
+      collection =>
+        typeof collection.id === "number" &&
+        !dataAppCollectionIDs.includes(collection.id),
+    );
```

Comment:
This is a temp workaround, the BE will hydrate app info on collections that are linked to apps, so the filter condition will become simpler